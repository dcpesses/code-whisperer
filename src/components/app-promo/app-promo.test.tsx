import {render} from '@testing-library/react';
import AppPromo from './index';

describe('AppPromo', () => {
  test('Should render as expected', () => {
    const {container} = render(<AppPromo />);
    expect(container).toMatchSnapshot();
  });
});
