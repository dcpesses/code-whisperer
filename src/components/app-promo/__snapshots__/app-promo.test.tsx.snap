// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppPromo > Should render as expected 1`] = `
<div>
  <div
    class="container-fluid app-promo"
  >
    <div
      class="row align-items-start justify-content-center mb-4"
    >
      <div
        class="col-sm-10 max-w-md-340"
      >
        <div
          class="pb-1"
        >
          <h1
            class="fw-bold text-center text-md-start"
          >
            <span
              class="d-inline-block"
            >
              Play With Friends,
            </span>
             
            <span
              class="d-inline-block"
            >
              Not Your Foes!
            </span>
          </h1>
          <div
            class="text-body fs-6 my-4 text-center text-md-start"
          >
            Developed as a way to help streamers thwart Jackbox trolls, this tool gives Twitch streamers the power to decide who can play in games they host by sending invites only to approved users.
          </div>
          <h3>
            How It Works:
          </h3>
          <ul
            class="text-body ps-4 fs-6 my-4"
          >
            <li
              class="mb-1"
            >
              Followers in chat who type 
              <b>
                !join
              </b>
               get added to the 
              <i>
                Interested
              </i>
               lobby.
            </li>
            <li
              class="mb-1"
            >
              The streamer then adds any or all of them to the 
              <i>
                Playing
              </i>
               section. You can even randomize who gets added too!
            </li>
            <li
              class="mb-1"
            >
              Once you enter the room code, press 
              <i>
                Send to Queue
              </i>
               to whisper everyone selected to play.
            </li>
          </ul>
        </div>
      </div>
      <div
        class="max-w-340 text-center"
      >
        <div
          class="twitch-chat-screenshot text-right"
        >
          <img
            alt="Twitch Chat"
            class="twitch-chat img-fluid shadow border border-secondary rounded-4 mx-auto mb-3"
            src="/code-whisperer/src/assets/twitch-chat.png"
          />
        </div>
      </div>
    </div>
    <div
      class="col-md-auto text-center"
    >
      <div
        class="browser-screenshot rounded-1 mb-3"
      >
        <div
          class="row g-1 g-sm-2 browser-bar text-start p-1 fs-6"
        >
          <div
            class="col-auto"
          >
            <i
              class="bi bi-arrow-left"
            />
            <i
              class="bi bi-arrow-clockwise"
            />
          </div>
          <div
            class="col"
          >
            <div
              class="browser-address-bar rounded-4 d-flex justify-content-between"
            >
              <div
                class="url-left"
              >
                <i
                  class="bi bi-info-circle"
                />
                <span
                  class="browser-address-url "
                >
                  http://
                  <span>
                    dcpesses.github.io
                  </span>
                  /code-whisperer
                </span>
              </div>
              <div
                class="url-right"
              >
                <i
                  class="bi bi-star"
                />
              </div>
            </div>
          </div>
          <div
            class="col-auto"
          >
            <i
              class="bi bi-three-dots-vertical"
            />
          </div>
        </div>
        <img
          alt="App Screenshot"
          class="screenshot img-fluid shadow border border-secondary rounded-2 mx-auto mb-1"
          src="/code-whisperer/src/assets/screenshot2.png"
        />
      </div>
    </div>
    <div
      class="col-md-auto text-center"
    >
      <div
        class="fs-6 my-3"
      >
        Oh, and did we mention it's free to use?
        <a
          href="#disclaimer"
        >
          <sup>
            *
          </sup>
        </a>
      </div>
      <a
        class="btn btn-sm fs-4 py-2 px-3 rounded-4 my-4 btn-purple bg-gradient focus-ring"
        href="https://id.twitch.tv/oauth2/authorize?client_id=VITE_APP_TWITCH_CLIENT_ID&response_type=code&scope=chat:read chat:edit moderation:read user:manage:whispers moderator:read:chatters user:read:moderated_channels channel:read:vips channel:read:editors moderator:manage:announcements user:read:subscriptions&redirect_uri=VITE_APP_REDIRECT_URI"
      >
        <span
          class="d-inline-block"
        >
          Go ahead, log in with 
          <strong>
            Twitch
          </strong>
        </span>
         
        <span
          class="d-inline-block"
        >
          and take it for a spin!
        </span>
      </a>
    </div>
    <div
      class="col-md-8 m-auto text-center"
    >
      <div
        class="fs-6 my-4"
      >
        View 
        <a
          class="emphasis"
          href="https://github.com/dcpesses/code-whisperer"
          rel="noreferrer"
          target="_blank"
        >
          Code Whisperer on GitHub
        </a>
         for more information, including documentation, planned features, technical support, and ways to contribute.
      </div>
    </div>
    <div
      class="col-md-auto text-center"
    >
      <div
        class="text-body fs-6 my-4"
      >
        <a
          name="disclaimer"
        >
          <sup>
            *
          </sup>
        </a>
        Yes seriously, it is free. That said, 
        <span
          class="d-inline-block"
        >
          you're also welcome to 
          <a
            class="emphasis"
            href="https://ko-fi.com/V7V6VSUT1"
            rel="noreferrer"
            target="_blank"
          >
            leave a tip!
          </a>
        </span>
        <div
          class="py-2"
        >
          <a
            class="emphasis"
            href="https://ko-fi.com/V7V6VSUT1"
            rel="noreferrer"
            target="_blank"
          >
            <button
              class="kofi-button btn btn-primary btn-sm"
              type="button"
            >
              <img
                alt="Ko-fi logo"
                class="img-fluid"
                src="data:image/svg+xml,%3csvg%20width='241'%20height='194'%20viewBox='0%200%20241%20194'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cmask%20id='mask0_1_219'%20style='mask-type:luminance'%20maskUnits='userSpaceOnUse'%20x='-1'%20y='0'%20width='242'%20height='194'%3e%3cpath%20d='M240.469%200.958984H-0.00585938V193.918H240.469V0.958984Z'%20fill='white'/%3e%3c/mask%3e%3cg%20mask='url(%23mask0_1_219)'%3e%3cpath%20d='M96.1344%20193.911C61.1312%20193.911%2032.6597%20178.256%2015.9721%20149.829C1.19788%20124.912%20-0.00585938%2097.9229%20-0.00585938%2067.7662C-0.00585938%2049.8876%205.37293%2034.3215%2015.5413%2022.7466C24.8861%2012.1157%2038.1271%205.22907%2052.8317%203.35378C70.2858%201.14271%2091.9848%200.958984%20114.545%200.958984C151.259%200.958984%20161.63%201.4088%20176.075%202.85328C195.29%204.76026%20211.458%2011.932%20222.824%2023.5955C234.368%2035.4428%20240.469%2051.2624%20240.469%2069.3627V72.9994C240.469%20103.885%20219.821%20129.733%20191.046%20136.759C188.898%20141.827%20186.237%20146.871%20183.089%20151.837L183.006%20151.964C172.869%20167.632%20149.042%20193.918%20103.401%20193.918H96.1281L96.1344%20193.911Z'%20fill='white'/%3e%3cpath%20d='M174.568%2017.9772C160.927%2016.6151%20151.38%2016.1589%20114.552%2016.1589C90.908%2016.1589%2070.9008%2016.387%2054.7644%2018.4334C33.3949%2021.164%2015.2058%2037.5285%2015.2058%2067.7674C15.2058%2098.0066%2016.796%20121.422%2029.0741%20142.107C42.9425%20165.751%2066.1302%20178.707%2096.1412%20178.707H103.414C140.242%20178.707%20160.25%20159.156%20170.253%20143.698C174.574%20136.874%20177.754%20130.058%20179.801%20123.234C205.947%20120.96%20225.27%2099.3624%20225.27%2072.9941V69.3577C225.27%2040.9432%20206.631%2021.164%20174.574%2017.9772H174.568Z'%20fill='white'/%3e%3cpath%20d='M15.1975%2067.7674C15.1975%2037.5285%2033.3866%2021.164%2054.7559%2018.4334C70.8987%2016.387%2090.906%2016.1589%20114.544%2016.1589C151.372%2016.1589%20160.919%2016.6151%20174.559%2017.9772C206.617%2021.1576%20225.255%2040.937%20225.255%2069.3577V72.9941C225.255%2099.3687%20205.932%20120.966%20179.786%20123.234C177.74%20130.058%20174.559%20136.874%20170.238%20143.698C160.235%20159.156%20140.228%20178.707%20103.4%20178.707H96.1264C66.1155%20178.707%2042.9277%20165.751%2029.0595%20142.107C16.7814%20121.422%2015.1912%2098.4563%2015.1912%2067.7674'%20fill='%23202020'/%3e%3cpath%20d='M32.2469%2067.9899C32.2469%2097.3168%2034.0654%20116.184%2043.6127%20133.689C54.5225%20153.924%2074.3018%20161.653%2096.8117%20161.653H103.857C133.411%20161.653%20147.736%20147.329%20155.693%20134.829C159.558%20128.462%20162.966%20121.417%20164.784%20112.547L166.147%20106.864H174.332C192.521%20106.864%20208.208%2092.09%20208.208%2073.2166V69.8082C208.208%2048.6669%20195.024%2037.5228%20172.058%2034.7987C159.102%2033.6646%20151.372%2033.2084%20114.538%2033.2084C89.7602%2033.2084%2072.0272%2033.4364%2058.6152%2035.4828C39.7483%2038.2134%2032.2407%2048.8951%2032.2407%2067.9899'%20fill='white'/%3e%3cpath%20d='M166.158%2083.6801C166.158%2086.4107%20168.204%2088.4572%20171.841%2088.4572C183.435%2088.4572%20189.802%2081.8619%20189.802%2070.9523C189.802%2060.0427%20183.435%2053.2195%20171.841%2053.2195C168.204%2053.2195%20166.158%2055.2657%20166.158%2057.9963V83.6866V83.6801Z'%20fill='%23202020'/%3e%3cpath%20d='M54.5321%2082.3198C54.5321%2095.732%2062.0332%20107.326%2071.5807%20116.424C77.9478%20122.562%2087.9515%20128.93%2094.7685%20133.022C96.8147%20134.157%2098.8611%20134.841%20101.136%20134.841C103.866%20134.841%20106.134%20134.157%20107.959%20133.022C114.782%20128.93%20124.779%20122.562%20130.919%20116.424C140.694%20107.332%20148.195%2095.7383%20148.195%2082.3198C148.195%2067.7673%20137.286%2054.8115%20121.599%2054.8115C112.28%2054.8115%20105.912%2059.5882%20101.136%2066.1772C96.8147%2059.582%2090.2259%2054.8115%2080.9001%2054.8115C64.9855%2054.8115%2054.5256%2067.7673%2054.5256%2082.3198'%20fill='%23FF5A16'/%3e%3c/g%3e%3c/svg%3e"
                style="height: 1em;"
              />
              <span
                class="px-2 align-middle"
              >
                Support me on Ko-fi 
              </span>
              <i
                aria-hidden="true"
                class="bi bi-box-arrow-up-right small"
              />
              <span
                class="visually-hidden"
              >
                 (link opens a new window)
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
